server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java

```diff
@@ -90,6 +90,9 @@ public Symbol normalizeSymbol(Function symbol, TransactionContext txnCtx, NodeCo
         if (arg instanceof Input<?> input) {
             return Literal.of(input.value() == null);
         }
+        if (arg instanceof Reference ref && !ref.isNullable()) {
+            return Literal.of(false);
+        }
```

Comment:
Testcases like `testSelectWhereVersionIsNullPredicate()` fails because `WhereClauseValidator` happens _after_ the non-null ref is evaluated.